src/tarball.py

```diff
@@ -59,25 +63,22 @@ def _update_repo(self, config):
         kernelci.build.update_repo(config, self._kdir)
         self.log.info("Repo updated")
 
-    def _make_tarball(self, config, describe):
-        name = '-'.join(['linux', config.tree.name, config.branch, describe])
-        tarball = f"{name}.tar.gz"
-        self.log.info(f"Making tarball {tarball}")
-        output_path = os.path.relpath(self._output, self._kdir)
-        cmd = """\
-set -e
-cd {kdir}
-git archive --format=tar --prefix={name}/ HEAD | gzip > {output}/{tarball}
-""".format(kdir=self._kdir, name=name, output=output_path, tarball=tarball)
+    def _make_tarball(self, target_dir, tarball_name):
+        self.log.info(f"Making tarball {tarball_name}")
+        tarball_path = os.path.join(self._output, f"{tarball_name}.tar.gz")
+        cmd = self.TAR_CREATE_CMD.format(
+            target_dir=target_dir,
+            prefix=tarball_name,
+            tarball_path=tarball_path
+        )
         self.log.info(cmd)
         kernelci.shell_cmd(cmd)
         self.log.info("Tarball created")
-        return tarball
+        return tarball_path
 
-    def _push_tarball(self, config, describe):
-        tarball_name = self._make_tarball(config, describe)
-        tarball_path = os.path.join(self._output, tarball_name)
-        self.log.info(f"Uploading {tarball_path}")
+    def _push_tarball(self, tarball_path):
+        tarball_name = os.path.basename(tarball_path)
+        self.log.info(f"Uploading {tarball_name}")
         tarball_url = self._storage.upload_single((tarball_path, tarball_name))
```

Comment:
Please correct me if I'm wrong: `tarball_path` now also contains `tarball_name` (see L68). Should the `upload_single()` call be adjusted?

TODO(pawiecz): double check if upload was unaffected/needs adjusting

Reply:
FYI The Python abstract class for Storage needs to be reviewed anyway as the `upload()` arguments are a bit hard to grasp with 2-tuples etc.  It was just a way to get something to work and tick a box.

Reply:
@gctucker right, thanks for clarifying. Taking point to look at it in the future, resolving thread for now.

Also, my initial comment made little sense: `tarball_path` *always* contained `tarball_name`, it was just processed differently.